### ISSUE TITLE
Additional option for the volume applet

### ISSUE THREAD

**Issue by thellamageek** [solus-project/budgie-desktop#626]
**Operating System**: Solus 1.2.0.5
**Budgie version**: 10.2.7

**Aditional option for the volume applet**

Ikei Hello, how are you ?, Well I tell you, in kde there an option in sound applet to adjust the volume to 150% and this in turn allows using the keyboard keys to adjust the volume to that 150%, this can be achieved in budgie / gnome to go to settings -> sound and do the ajust, sadly to lower the volume with the keys "reset" settings and only allows upload 100% i personally use this option much because one my jobs is to make transcripts and sometimes the sound is very low as well as sometimes the sound is very high and is a little uncomfortable to be entering settings all the time, which very kindly ask is that there is an option in raven or the same sound applet to make it as in kde. Whatever be the desicion, again thank you very much for the work with budgie and solus.

here the option in kde: (sorry for the spanish) haha

![screenshot_20160926_130951](https://cloud.githubusercontent.com/assets/22453960/18844431/37762a6c-83eb-11e6-94f6-d078576dbd87.png)

**Comment by haevalencia** [comment]
It seems a good choice for Raven.

**Comment by tannji** [comment]
I also run into situations where this would be handy and more convenient.  Used it frequently in Mint, which I dropped.

**Comment by haevalencia** [comment]
It would also be useful to add a volume control for the input system. Today it is very common to use VoIP applications, send voice messages or record audio or video for different circumstances.

**Comment by mussel** [comment]
I think this feature would be very useful. 
We could do this similar to how Ubuntu does it. We could add a checkbox to the System's Sound settings to allow for the sound to be louder than 100%. 



![selection_003](https://cloud.githubusercontent.com/assets/1262461/19207939/4a73898c-8ccb-11e6-9ffc-b04b7b38e67c.png)
If checked, the system would treat 150% as the new max. volume and thus allowing you to change the volume to above and bellow 100% normally, without the resetting behavior.

PS: I would be willing to implement it.

**Comment by ikeydoherty** [comment]
We don't own System Settings, that's gnome-control-center. Has to go somewhere else or we roll our own settings app.

**Comment by mussel** [comment]
I was under the impression that we could apply some patches to gnome-control-center, during packaging, to add this option. Would it be possible?

Another place to put this option would be in the popover for the sound applet; perhaps underneath progress bar. However, this would probably require making the popover wider in order to fit the text to "explain" the option.

**Comment by ikeydoherty** [comment]
We need to think in terms of having it agnostic @mussel  - that would only work for Solus, not Budgie :)

**Comment by mussel** [comment]
I had not thought about that. =] 

So what about putting the option in the sound applet popover? What do you think? Do you have any other suggestions?

**Comment by mussel** [comment]
What about putting the amplification option within the Raven's SoundWidget, like so:

![selection_004](https://cloud.githubusercontent.com/assets/1262461/19255706/1330203e-8f37-11e6-97b6-556b85b7f97c.png)

**Comment by AntiNSA** [comment]
I really need this feature. This is a big turn off in gnome for me vs ubuntu. Has anyone found a solution?

**Comment by orschiro** [comment]
I like @mussel's proposal to put the amplification option into the sidebar for the special circumstances when it is needed. Not necessary to add it to the applet icon, in my opinion.

Visually turn the bar from blue to red if the users passes 100%.

What do others think?

**Comment by mussel** [comment]
TL;DR: Working on issue this issue. Need help making mediakeys set volume louder than 100%.
Questions:

1. Is budgie using gnome-settings-daemon's media-keys plugin for handling the mediakey functionality?
2. If so, how does it do that?
3. If not, how are mediakeys handled?

@ikeydoherty , I've been working on this but I need your help in something. I already implemented the functionality for setting the sound louder than 100% in both the Raven's SoundWidget and the SoundIndicator from the Status applet. Now, I only need to enable the functionality when using the mediakeys. The problem is, I have no idea how these mediakeys are handled by budgie.

I searched the git history for mentions to 'mediakey' and it pointed me to the file 'src/wm/shim.vala'. I really did not understand what this shim is doing and how it relates to the mediakeys. My current guess is that the key events are (somehow) captured by gnome-session and it (somehow) handles the volume change. I looked on google for some more information, but the only thing I could find was this thing called gnome-settings-daemon that has a 'media-keys' plugin that listens to mediakey presses and adjusts the volume accordingly.

So I guess my questions are:
1. Is budgie using gnome-settings-daemon's 'media-keys' plugin for handling the mediakey functionality?
2. If so, how does it do that?
3. If not, how are mediakeys handled?

**Comment by orschiro** [comment]
@mussel not providing any useful comment in response to your question but just wanted to thank you so much for working on this! I needed this for so long and I am so glad it's soon arriving. Thanks, really!

**Comment by codegreens** [comment]
I second this as something that'd be a great enhancement to Budgie

my laptop speakers aren't the best, and every single time I watch a video I find myself opening settings and sliding the volume past 100%. using the mediakeys would be much smoother

**Comment by JoshStrobl** [comment]
Resolved as of c61b2210a870ce4380920581aedb6cede6e98e30
